src/simulation/LoadGenerator.cpp

```diff
@@ -378,20 +378,23 @@ LoadGenerator::start(GeneratedLoadConfig& cfg)
     if (cfg.mode != LoadGenMode::PAY_PREGENERATED)
     {
         // For upgrade modes, use root account (represented by special ID)
+        uint32_t accounts = cfg.nAccounts;
         if (cfg.mode == LoadGenMode::SOROBAN_UPGRADE_SETUP ||
             cfg.mode == LoadGenMode::SOROBAN_CREATE_UPGRADE)
         {
             mAccountsAvailable.insert(TxGenerator::ROOT_ACCOUNT_ID);
-        }
-        else
-        {
-            // Mark all accounts "available" as source accounts
-            for (auto i = 0u; i < cfg.nAccounts; i++)
+            if (accounts)
```

Comment:
Nit: accounts == 0

Reply:
This evaluates to accounts != 0 anyway, so I'd prefer to keep it as-is to avoid delaying the merge further.